src/v/cluster/leader_routing_frontend.h

```diff
@@ -0,0 +1,248 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#pragma once
+
+#include "cluster/logger.h"
+#include "cluster/members_table.h"
+#include "cluster/metadata_cache.h"
+#include "cluster/partition_leaders_table.h"
+#include "cluster/partition_manager.h"
+#include "cluster/shard_table.h"
+#include "cluster/types.h"
+#include "model/metadata.h"
+#include "model/timeout_clock.h"
+#include "rpc/connection_cache.h"
+#include "seastarx.h"
+#include "vformat.h"
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/sharded.hh>
+
+namespace rpc {
+class connection_cache;
+} // namespace rpc
+
+namespace cluster {
+
+// Encapsulates a frontend that aims to direct RPCs to a specific NTP's leader.
```

Comment:
nit on naming: Is there an equivalent "backend" ? Wondering why this is a "leader_routing_frontend" instead of say a "leader_router"

Reply:
Done